core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +64,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
```

Comment:
as far as I can tell we are scheduleing on a fixed delay every 10 seconds so I wonder if we should just use this and don't schedule another task? I might be missing something